### ISSUE TITLE
tests fail when building with HTSLIB_MODE=external

### ISSUE THREAD

**Issue by nieder** [pysam-developers/pysam#665]
When building pysam-0.14.1 against my locally installed htslib-1.7, the tests fail like this:

```python
	export HTSLIB_MODE=external HTSLIB_INCLUDE_DIR=/sw/include HTSLIB_LIBRARY_DIR=/sw/lib
	/sw/bin/python2.7 setup.py build
# pysam: cython is available - using cythonize if necessary
# pysam: htslib mode is external
# pysam: HTSLIB_CONFIGURE_OPTIONS=None
...
/sw/bin/pytest-2.7 tests
...
===================================================================================== FAILURES ======================================================================================
__________________________________________________________________________ TestRemoteFileHTTP.testFetchAll __________________________________________________________________________

self = <tabix_test.TestRemoteFileHTTP testMethod=testFetchAll>

    def setUp(self):
>       if not pysam.config.HAVE_LIBCURL or not check_url(self.url):
E       AttributeError: 'module' object has no attribute 'HAVE_LIBCURL'

tests/tabix_test.py:1049: AttributeError
___________________________________________________________________________ TestRemoteFileHTTP.testHeader ___________________________________________________________________________

self = <tabix_test.TestRemoteFileHTTP testMethod=testHeader>

    def setUp(self):
>       if not pysam.config.HAVE_LIBCURL or not check_url(self.url):
E       AttributeError: 'module' object has no attribute 'HAVE_LIBCURL'

tests/tabix_test.py:1049: AttributeError
_____________________________________________________________________ TestRemoteFileHTTPWithHeader.testFetchAll _____________________________________________________________________

self = <tabix_test.TestRemoteFileHTTPWithHeader testMethod=testFetchAll>

    def setUp(self):
>       if not pysam.config.HAVE_LIBCURL or not check_url(self.url):
E       AttributeError: 'module' object has no attribute 'HAVE_LIBCURL'

tests/tabix_test.py:1088: AttributeError
______________________________________________________________________ TestRemoteFileHTTPWithHeader.testHeader ______________________________________________________________________

self = <tabix_test.TestRemoteFileHTTPWithHeader testMethod=testHeader>

    def setUp(self):
>       if not pysam.config.HAVE_LIBCURL or not check_url(self.url):
E       AttributeError: 'module' object has no attribute 'HAVE_LIBCURL'

tests/tabix_test.py:1088: AttributeError
================================================================================= warnings summary ==================================================================================
tests/AlignmentFile_test.py::TestTruncatedBAM::testTruncatedBam2
  /sw/build.build/pysam-py27-0.14.1-1/pysam-0.14.1/tests/AlignmentFile_test.py:1397: UserWarning: no BGZF EOF marker; file may be truncated
    ignore_truncation=True)

-- Docs: http://doc.pytest.org/en/latest/warnings.html
========================================================== 4 failed, 891 passed, 15 skipped, 1 warnings in 153.23 seconds ===========================================================
```
My external htslib-1.7 was configured with `--enable-libcurl` and `HAVE_LIBCURL` was defined in its `config.h` file. If I switch the build to htslib mode = shared, then the tests all pass.

**Comment by 0xaf1f** [comment]
I'm seeing this as well when trying to update the Debian package.

**Comment by smoe** [comment]
I happily prepare a VM for you should you find any difficulty to reproduce this behaviour.

**Comment by AndreasHeger** [comment]
Thanks, that might be useful, afaik we check external mode in our ci on
Travis. Cheers, Andreas

**Comment by AndreasHeger** [comment]
But have not checked yet, until tonight.

**Comment by jmarshall** [comment]
I haven't recently run the tests with `HTSLIB_MODE=external`, but this should have been fixed by 2cc584b574afef004fb6cb6554ab74145259f8fe.
